### PR TITLE
Spectrum analyser

### PR DESCRIPTION
**Pull request by rivkabr**

### CODE REVIEW

**Comment by rivkabr** [comment]
[Spectrum analyser POC](https://trello.com/c/9Y4aQPrT/3108-spectrum-analyser-poc)

File: src/cljs/audyx/audio/spectrum_analyser.cljs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        sum-all (sum-all-octaves fft-array fft-size analyser)]
+    (set-octaves-fn (clj->js octaves-volume-array))
+    (set-sum-all-fn sum-all)
+    (->
+      (js/requestAnimationFrame #(read-input-values analyser set-octaves-fn set-sum-all-fn set-request octaves-band fft-size))
+      (set-request))))
+
+(defn start-read-microphone [device-id set-octaves-fn set-sum-all-fn set-request]
+  (go
+    (match (<! (get-media-stream {:audio {:echoCancellation false, :autoGainControl false, :noiseSuppression false} }))
```
Comment (tzipisheinin) [inline src/cljs/audyx/audio/spectrum_analyser.cljs]: Add let constrains

File: src/cljs/audyx/audio/spectrum_analyser.cljs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        max-index (stream/calc-max-index max-frequency (audio-context/sample-rate) fft-size analyser)
+        min-index (stream/calc-min-index min-frequency (audio-context/sample-rate) fft-size)]
+    (calc-volume fft-array min-index max-index)))
+
+(def all-octaves
+  (map #(Math/pow 10 (* 0.1 %)) (range 23 42)))
+
+(defn octave-band [fft-size analyser octave]
+  (let [fd (Math/pow 10 0.05)
+        fupper (* octave fd)
```
Comment (tzipisheinin) [inline src/cljs/audyx/audio/spectrum_analyser.cljs]: f-upper

File: src/cljs/audyx/audio/spectrum_analyser.cljs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+   (->>
+     (max-dbspl)
+     (audio-context/dbfs->dbspl volume)))
+
+(defn calc-volume [fft-array min-index max-index]
+  (as->
+    (stream/calc-volume fft-array min-index max-index) $
+    (dbfs->dbspl $)))
+
+(defn sum-octave [fft-array {:keys [octave min-index max-index]}]
```
Comment (tzipisheinin) [inline src/cljs/audyx/audio/spectrum_analyser.cljs]:
```suggestion
(defn octave-sum [fft-array {:keys [octave min-index max-index]}]
```

File: src/cljs/audyx/audio/spectrum_analyser.cljs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+(defn read-input-values [analyser set-octaves-fn set-sum-all-fn set-request octaves-band fft-size]
+  (let [fft-array (audio-context/fft-data-array analyser)
+        octaves-volume-array (map (partial sum-octave fft-array) octaves-band)
+        sum-all (sum-all-octaves fft-array fft-size analyser)]
+    (set-octaves-fn (clj->js octaves-volume-array))
+    (set-sum-all-fn sum-all)
+    (->
+      (js/requestAnimationFrame #(read-input-values analyser set-octaves-fn set-sum-all-fn set-request octaves-band fft-size))
+      (set-request))))
```
Comment (tzipisheinin) [inline src/cljs/audyx/audio/spectrum_analyser.cljs]:
```suggestion
      (set-request-fn))))
```

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@audyx/audyx-deps",
-  "version": "1.0.47",
+  "version": "1.0.56-spectrum",
```
Comment (tzipisheinin) [inline package.json]: update version

File: src/cljs/audyx/audio/spectrum_analyser.cljs (hunk 2 above)
Comment (tzipisheinin) [reply src/cljs/audyx/audio/spectrum_analyser.cljs]: done

File: src/cljs/audyx/audio/spectrum_analyser.cljs (hunk 3 above)
Comment (tzipisheinin) [reply src/cljs/audyx/audio/spectrum_analyser.cljs]: done

File: src/cljs/audyx/audio/spectrum_analyser.cljs (hunk 4 above)
Comment (tzipisheinin) [reply src/cljs/audyx/audio/spectrum_analyser.cljs]: done

File: src/cljs/audyx/audio/spectrum_analyser.cljs (hunk 1 above)
Comment (tzipisheinin) [reply src/cljs/audyx/audio/spectrum_analyser.cljs]: done

**Pull request merged**